sql/sqlwire/sql_api.proto

```diff
@@ -56,27 +56,48 @@ message SQLResponseHeader {
 }
 
 message Datum {
-  option (gogoproto.onlyone) = true;
-  oneof value {
-    bool bval = 1;
-    int64 ival = 2;
-    double dval = 3;
-    bytes blobval = 4;
+  enum Kind {
+    NULL = 0;
+    // Fixed width, big-endian. Also used for boolean values.
+    INT64 = 1;
+    // Fixed width, big-endian.
+    FLOAT64 = 2;
+    // 8-byte big-endian seconds (since Jan 1, 1970) + 4-byte big-endian
+    // nanoseconds.
+    TIME = 3;
+    // Raw bytes.
+    BYTES = 4;
```

Comment:
This set is much more restricted than `ColumnType.Kind` and is what the user can pass as the value for a placeholder. Perhaps a different name should be used here. I'm open to suggestions.